phocoa/framework/widgets/WFView.php

```diff
@@ -287,6 +292,35 @@ public static function exposedProperties()
     }
 
     /**
+     * Set a key-value pair in the dataAttributes data set.
+     *
+     * @param string Key -- use only chars safe for use as an HTML attribute name.
+     * @param string Value -- will be HTML-encoded as an attribute.
+     */
+    public function setDataAttribute($k, $v)
+    {
+        $this->dataAttributes[$k] = $v;
+    }
+
+    /**
+     * @return string All key-value pairs of the dataAttributes data set rendered out for inclusion in an HTML entity. All keys are prefixed with 'data-'
+     */
+    protected function renderDataAttributesToHTML()
+    {
+        if (count($this->dataAttributes) === 0)
+        {
+            return '';
+        }
+
+        $html = ' ';
```

Comment:
Why is the empty space needed? `$html = ' '`.  There's already padding at line 317, at the start and end of quotes.